### ISSUE TITLE
Divergent implicits when using `cats.implicits._`

### ISSUE THREAD

**Issue by gvolpe** [neotypes/neotypes#36]
I've read #35 and I understand it refers to the `import neotypes.Async._`? I am currently experiencing the same issue but also with `import neotypes.implicits._` using version `0.9.0`.

Here's the original code I had using the Neo4j java driver:

```scala
import cats.Applicative
import cats.effect._
import cats.effect.implicits._
import cats.implicits._
import org.neo4j.driver.v1._

object Neo {

  case class NeoConfig(uri: String, user: String, pass: String)

  // TODO: do proper logging
  def putStrLn[F[_]: Sync, A](a: A): F[Unit] = Sync[F].delay(println(a))

  private def loadConfig[F[_]: Applicative]: F[NeoConfig] =
    NeoConfig("bolt://localhost:7687", "neo4j", "test").pure[F]

  def mkClient[F[_]: Sync]: Resource[F, NeoClient[F]] =
    Resource.make[F, NeoClient[F]](
      loadConfig[F].flatMap { c =>
        putStrLn("NeoClient[F] >> Connecting to Neo4j...") *>
          Sync[F]
            .delay(GraphDatabase.driver(c.uri, AuthTokens.basic(c.user, c.pass)))
            .map(new LiveNeoClient(_))
            .widen[NeoClient[F]] <*
          putStrLn("NeoClient[F] >> Connected")
      }
    )(_.close)

}

trait NeoClient[F[_]] {
  def close: F[Unit]
  def transaction: Resource[F, Transaction]
}

class LiveNeoClient[F[_]: Sync] private[persistence] (driver: Driver) extends NeoClient[F] {
  def close: F[Unit] = Neo.putStrLn("NeoClient[F] >> Closing connection") *> Sync[F].delay(driver.close())
  def transaction: Resource[F, Transaction] =
    Resource.make(Sync[F].delay(driver.session.beginTransaction))(tx => Sync[F].delay(tx.close))
}
```

This compiles fine until I add any of the imports `neotypes.Async._` and/or `neotypes.implicits._`

```
[error] /workspace/.../Neo.scala:22:17: type mismatch;
[error]  found   : F[Neo.NeoConfig]
[error]  required: ?{def flatMap: ?}
[error] Note that implicit conversions are not applicable because they are ambiguous:
[error]  both method toFlatMapOps in trait ToFlatMapOps of type [F[_], C](target: F[C])(implicit tc: cats.FlatMap[F])cats.FlatMap.Ops[F,C]{type TypeClassType = cats.FlatMap[F]}
[error]  and method AsyncExt in object implicits of type [F[_], T](m: F[T])neotypes.implicits.AsyncExt[F,T]
[error]  are possible conversion functions from F[Neo.NeoConfig] to ?{def flatMap: ?}
[error]       loadConfig[F].flatMap { c =>
[error]                 ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 2 s, completed Jun 28, 2019 10:58:31 AM
```

Are there any known workarounds to this? I see there's been some work done in #33 but the error still seems to be there.

**Comment by BalmungSan** [comment]
No #35 refers to the same implicits you are importing _(thus this is a duplicate of that one)_.
The problem is with the `AsynSyntax` implicit which is included in the **all** & **syntax.all** objects.

#33 includes a workarround, and it is that you can just import everything but that.
However, as discussed in #35 that is far from ideal in an ergonomic sense.
That is why, after #34 gets merged, we will remove that from the public API, as such syntax is only useful to the project.
Again, just the syntax, the `Async[F[_]]` _typeclass_ itself, will remain public, ass hiding it would disallow users to implement their own instance for any other type.

**Comment by gvolpe** [comment]
Closing duplicated.
